### PR TITLE
provide docker file for github builds with cmake

### PR DESCRIPTION
**Pull request by pellcorp**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -102,7 +102,7 @@ libhv.a:
 
 libspdlog.a:
 	@mkdir -p $(SPDLOG_DIR)/build
-	@cmake -B $(SPDLOG_DIR)/build -S $(SPDLOG_DIR)/
+	@cmake -B $(SPDLOG_DIR)/build -S $(SPDLOG_DIR)/ -DCMAKE_CXX_COMPILER=$(CXX)
```
Comment (ballaswag) [inline Makefile:105]: Is this needed? Are line 5 and 6 where CC/CXX are defined no enough?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM ubuntu:22.04
```
Comment (ballaswag) [inline Dockerfile:1]: Maybe just do `FROM ballaswag/guppydev:latest` then ```RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
    apt-get install -y --no-install-recommends cmake 
    apt-get clean all && \
    apt-get -y autoremove```

Will try this and publish one with cmake

**Comment by pellcorp** [comment]
I am closing this, as it seems like you have it sorted.

File: Makefile (hunk 1 above)
Comment (pellcorp) [reply Makefile:105]: In my local environment I could not get cmake to work without it, but perhaps that was only on the command line.

**Pull request closed without merging**